src/I2C.cpp

```diff
@@ -0,0 +1,148 @@
+//
+// Created by noeel on 05-10-21.
+//
+//#include "libMPSSE_i2c.h"
+#include "I2C.h"
+#include <mpsse.h>
+#include <malloc.h>
+#include <memory.h>
+#include <stdlib.h>
+#include <system_error>
+#include <cassert>
+#include "cli.h"
+
+uint8 I2C::i2c_write(uint8 address, uint8 reg, uint8 *data, size_t data_size) {
+    size_t write_size = 2 + data_size; // address + reg + datasize
+    char *writebuffer = (char *) malloc(write_size);
+    uint8 ret = 0;
+    writebuffer[0] = ((address << 1) | 0b00000000);
+    writebuffer[1] = reg;
+    for (int i = 0; i < data_size; ++i) {
+        writebuffer[i + 2] = data[i];
+    }
+    if (Start(mpsse) == MPSSE_OK) {
+        if (Write(mpsse, writebuffer, write_size) == MPSSE_OK) {
+            if (GetAck(mpsse) == ACK)
+                ret = 0;
+            else
+                ret = 1;
+        }
+    }
+    if (Stop(mpsse) != MPSSE_OK)
+        ret = 2;
+    free(writebuffer);
+    return ret;
+}
+
+
+/**
+ * Read data from I2C device
+ * @param address address of slave
+ * @param reg register to read
+ * @param buffer store read data here
+ * @param buffer_size read size
+ * @return 0 if success, 1 if no response and 2 if hardware failed.
+ */
```

Comment:
These comment styles are great. It would be awesome to have these throughout the classes.

Reply:
:+1: